listener.c

```diff
@@ -228,8 +231,14 @@ evconnlistener_new_bind(struct event_base *base, evconnlistener_cb cb,
 	if (fd == -1)
 		return NULL;
 
-	if (setsockopt(fd, SOL_SOCKET, SO_KEEPALIVE, (void*)&on, sizeof(on))<0)
-		goto err;
+#if defined(_WIN32) && defined(EVENT__HAVE_AFUNIX_H)
+	if (!evutil_check_working_afunix_()) {
+#endif
+		if (setsockopt(fd, SOL_SOCKET, SO_KEEPALIVE, (void*)&on, sizeof(on))<0)
```

Comment:
But even in case AF_UNIX **is supported**, we still can have AF_INET socket, or I'm missing something?